logic/src/commonTest/kotlin/com/wire/kalium/logic/data/message/MessageMapperTest.kt

```diff
@@ -0,0 +1,104 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.kalium.logic.data.message
+
+import com.wire.kalium.logic.data.user.UserId
+import com.wire.kalium.persistence.dao.QualifiedIDEntity
+import com.wire.kalium.persistence.dao.message.DeliveryStatusEntity
+import com.wire.kalium.persistence.dao.message.MessageEntity
+import com.wire.kalium.persistence.dao.message.MessageEntityContent
+import com.wire.kalium.persistence.dao.reaction.ReactionsEntity
+import kotlinx.datetime.Instant
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class MessageMapperTest {
+
+    val arrangement = Arrangement()
+
+    @Test
+    fun givenRegularMessageEntityWithDeliveredStatus_whenMapping_thenTheMessageHasExpectedData() {
+        // given / when
+        val result = arrangement.withRegularMessage(
+            status = MessageEntity.Status.DELIVERED
+        )
+
+        // then
+        assertEquals(result.status, Message.Status.Delivered)
+    }
+
+    @Test
+    fun givenRegularMessageEntityWithReadStatus_whenMapping_thenTheMessageHasExpectedData() {
+        // given / when
+        val result = arrangement.withRegularMessage(
+            status = MessageEntity.Status.READ,
+            readCount = 10
+        )
+
+        // then
+        assertEquals(result.status, Message.Status.Read(10))
+    }
```

Comment:
Why only these two statuses are tested? If they only test mapping statuses properly then shouldn't `MessageEntity.Status.toModel` be used in tests instead of whole `fromEntityToMessage`? And what about testing the other way around - from `Message.Status` to `MessageEntity.Status`? 😄

Reply:
I mean I could use directly toModel() but I decided to make it as such to make it easier for future test covering it uses it under the hood anyways. 

I will add a test to test it another way around 👍